tests/units/minter/test_public_buy.cairo

```diff
@@ -116,7 +116,57 @@ func test_buy_revert_not_enough_available_value{
     %{ mock_call(context.mocks.carbonable_project_address, "totalValue", [0, 0]) %}
     %{ mock_call(context.mocks.payment_token_address, "transferFrom", [1]) %}
     %{ expect_revert("TRANSACTION_FAILED", "CarbonableMinter: not enough available value") %}
-    CarbonableMinter.public_buy(value);
+    CarbonableMinter.public_buy(value=value, force=FALSE);
+    %{ stop() %}
+    return ();
+}
+
+@external
+func test_force_buy_over_min{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr}() {
+    alloc_locals;
+
+    // prepare minter instance
+    let (local context) = prepare(
+        public_sale_open=TRUE,
+        max_value_per_tx=5,
+        min_value_per_tx=1,
+        max_value=10,
+        unit_price=10 * 10 ** 6,
+        reserved_value=9,
+    );
+
+    // run scenario
+    %{ stop=start_prank(context.signers.anyone) %}
+    let value = 2;
+    %{ mock_call(context.mocks.carbonable_project_address, "totalValue", [0, 0]) %}
+    %{ mock_call(context.mocks.carbonable_project_address, "mintNew", [1337, 0]) %}
+    %{ mock_call(context.mocks.payment_token_address, "transferFrom", [1]) %}
+    CarbonableMinter.public_buy(value=value, force=TRUE);
+    %{ stop() %}
+    return ();
+}
+
+@external
+func test_force_buy_below_min{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr}() {
```

Comment:
Maybe also test the case where the force is due to prebuys (this only deals with reserved value)

Reply:
Not required since it calls the same function under the hood